src/RoslynPad/MainViewModel.cs

```diff
@@ -233,9 +254,16 @@ public void CreateNewDocument()
 
         private async Task CloseCurrentDocument()
         {
-            if (CurrentOpenDocument != null)
+            await CloseDocument(CurrentOpenDocument).ConfigureAwait(false);
+        }
+
+        public async Task CloseAllDocuments()
+        {
+            // can't modify the collection while enumerating it.
```

Comment:
You can use `OpenDocuments.ToArray()`. Also remove the comment - this is very common knowledge in .NET :)